File: LeetCode-JavaScript/35-搜索插入位置.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var searchInsert = function(nums, target) {
    let low = 0
    let high = nums.length
    while (low < high){
        let mid = parseInt(low + (high-low)/2)
        if (nums[mid] < target){
            low = mid +1
        } else if (nums[mid] > target){
            high = mid
        } else {
            return mid
        }
    }
    return low
};

File: LeetCode-Go/35-搜索插入位置.go
package main

import "fmt"

func main() {
 a := []int{1,2,3,4,5,6}
 fmt.Println(searchInsert(a,5))
}

func searchInsert(nums []int, target int) int {
	low := 0
	high := len(nums)
	for ;low<high; {
		mid := low + (high - low)/2
		if nums[mid] > target{
			high = mid
		} else if nums[mid] < target {
			low = mid + 1
		} else {
			return mid
		}
	}
	return low
}
